runtime/tools/python/ttrt/common/golden.py

```diff
@@ -103,7 +103,7 @@ def get_pcc(golden, calculated):
     )
 
 
-def golden(binary, programContext, opContext):
+def golden(runtimeConfig, binary, programContext, opContext):
```

Comment:
It's unclear to me that we want to have `struct RuntimeConfig` be part of c++ / the runtime.  I think this might be better as just part of ttrt. i.e. we could pass a runtime_config via capture (or functools.partial).  This way the c++ doesn't have to be in the business of knowing about golden info / checks which would be confusing for registered callbacks that might not be doing golden comparison at all.

```python
def get_golden_fn(runtime_config, data):
    def golden(binary, programContext, opContext):
        ...
        data[loc] = {"atol": atol, "pcc": pcc, ...}
    return golden
```

It also might be better to keep this function out of the business of dealing with files, `data` could be passed in and it gets appended to.

Reply:
resolved after cherry-picking Collin's fix